linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java

```diff
@@ -0,0 +1,358 @@
+    return getConnection(propMap);
+  }
+
+  public Connection getConnection(Configuration configuration) {
+    String clusterConnUniqueKey = generateUniqueConnectionKey(configuration);
+    LOG.info("Start to get connection for cluster {}.", clusterConnUniqueKey);
+    try {
+      lock.lock();
+      if (!connectionMap.containsKey(clusterConnUniqueKey)) {
+        if (isKerberosAuthType(configuration) && kerberosEnvInit.compareAndSet(false, true)) {
+          doKerberosLogin(configuration);
+        }
+        Connection connection;
+        String proxyUser = getKerberosProxyUser(configuration);
+        UserGroupInformation kerberosLoginUser = UserGroupInformation.getLoginUser();
+        String kerberosLoginShortUserName = kerberosLoginUser.getShortUserName();
+        if (StringUtils.isNotBlank(proxyUser) && !proxyUser.equals(kerberosLoginShortUserName)) {
+          UserGroupInformation ugi =
+              UserGroupInformation.createProxyUser(proxyUser, kerberosLoginUser);
+          connection =
+              ugi.doAs(
+                  (PrivilegedAction<Connection>)
+                      () -> {
+                        try {
+                          return ConnectionFactory.createConnection(configuration);
+                        } catch (IOException e) {
+                          LOG.error(
+                              HBaseErrorCodeSummary.HBASE_CLIENT_CONN_CREATE_FAILED.getErrorDesc(),
+                              e);
+                          throw new JobExecutorException(
+                              HBaseErrorCodeSummary.HBASE_CLIENT_CONN_CREATE_FAILED.getErrorCode(),
+                              HBaseErrorCodeSummary.HBASE_CLIENT_CONN_CREATE_FAILED.getErrorDesc());
+                        }
+                      });
+          LOG.info("Successfully create a connection {} and proxy user {}", connection, proxyUser);
+        } else {
+          connection = ConnectionFactory.createConnection(configuration);
+          LOG.info("Successfully create a connection {}.", connection);
+        }
+        connectionMap.put(clusterConnUniqueKey, connection);
+        return connection;
+      }
+    } catch (IOException e) {
+      LOG.error(HBaseErrorCodeSummary.HBASE_CLIENT_CONN_CREATE_FAILED.getErrorDesc(), e);
+      throw new JobExecutorException(
+          HBaseErrorCodeSummary.HBASE_CLIENT_CONN_CREATE_FAILED.getErrorCode(),
+          HBaseErrorCodeSummary.HBASE_CLIENT_CONN_CREATE_FAILED.getErrorDesc());
+    } finally {
+      lock.unlock();
+    }
+    return connectionMap.get(clusterConnUniqueKey);
+  }
+
+  public Connection getConnection(Map<String, String> prop) {
+    Configuration configuration = buildConfiguration(prop);
+    return getConnection(configuration);
+  }
+
+  private void doKerberosLogin(Configuration configuration) {
+    String principal = configuration.get(KERBEROS_PRINCIPAL);
+    String keytab = configuration.get(KERBEROS_KEYTAB_FILE);
```

Comment:
Need to adapt Linkis's existing kerberos tools?

Reply:
I feel it is necessary to optimize it later.